controllers/ibmvpcmachinetemplate_controller_test.go

```diff
@@ -0,0 +1,210 @@
+		mockvpc := mock.NewMockVpc(gomock.NewController(t))
+		reconciler := &IBMVPCMachineTemplateReconciler{
+			Client: testEnv.Client,
+		}
+		return gomock.NewController(t), mockvpc, reconciler
+	}
+
+	t.Run("with valid profile ", func(tt *testing.T) {
+		g := NewWithT(tt)
+		var expectedCapacity corev1.ResourceList
+		profileDetails := vpcv1.InstanceProfile{
+			Name: pointer.String("bx2-4x16"),
+			VcpuCount: &vpcv1.InstanceProfileVcpu{
+				Type:  pointer.String("fixed"),
+				Value: pointer.Int64(4),
+			},
+			Memory: &vpcv1.InstanceProfileMemory{
+				Type:  pointer.String("fixed"),
+				Value: pointer.Int64(16),
+			},
+		}
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("namespace-%s", util.RandomString(5)))
+		VPCMachineTemplate := *stubVPCMachineTemplate("bx2-4x16")
+
+		expectedCapacity = map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceCPU:    resource.MustParse("4"),
+			corev1.ResourceMemory: resource.MustParse("16G"),
+		}
+		createObject(g, &VPCMachineTemplate, ns.Name)
+		defer cleanupObject(g, &VPCMachineTemplate)
+
+		mockController, mockvpc, reconciler := setup(t)
+		t.Cleanup(mockController.Finish)
+		g.Expect(err).To(BeNil())
+		defer func() {
+			g.Expect(testEnv.Cleanup(ctx, ns)).To(Succeed())
+		}()
+		mockvpc.EXPECT().GetInstanceProfile(gomock.AssignableToTypeOf(&vpcv1.GetInstanceProfileOptions{})).Return(&profileDetails, &core.DetailedResponse{}, nil)
+		_, err = reconciler.reconcileNormal(ctx, mockvpc, VPCMachineTemplate)
+		if err != nil {
+			t.Fatalf("ReconcileNormal is not expected to return an error, error: %v", err)
+		}
+		g.Expect(err).To(BeNil())
+		g.Eventually(func() bool {
+			machineTemplate := &infrav1beta2.IBMVPCMachineTemplate{}
+			key := client.ObjectKey{
+				Name:      VPCMachineTemplate.Name,
+				Namespace: ns.Name,
+			}
+			err = testEnv.Get(ctx, key, machineTemplate)
+			g.Expect(err).To(BeNil())
+			return reflect.DeepEqual(machineTemplate.Status.Capacity, expectedCapacity)
+		}, 10*time.Second).Should(Equal(true))
+
+	},
+	)
+
+	t.Run("with invalid profile ", func(tt *testing.T) {
+		g := NewWithT(tt)
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("namespace-%s", util.RandomString(5)))
+
+		VPCMachineTemplate := *stubVPCMachineTemplate("")
+		createObject(g, &VPCMachineTemplate, ns.Name)
+		defer cleanupObject(g, &VPCMachineTemplate)
+
+		mockController, mockvpc, reconciler := setup(t)
+		t.Cleanup(mockController.Finish)
+		g.Expect(err).To(BeNil())
+		defer func() {
+			g.Expect(testEnv.Cleanup(ctx, ns)).To(Succeed())
+		}()
+		mockvpc.EXPECT().GetInstanceProfile(gomock.AssignableToTypeOf(&vpcv1.GetInstanceProfileOptions{})).Return(nil, &core.DetailedResponse{}, nil)
+		_, err = reconciler.reconcileNormal(ctx, mockvpc, VPCMachineTemplate)
+		if err == nil {
+			t.Fatalf("ReconcileNormal is  expected to return an error")
+		} else {
+			g.Expect(err).NotTo(BeNil())
+		}
+	},
+	)
+}
+
+func stubVPCMachineTemplate(profile string) *infrav1beta2.IBMVPCMachineTemplate {
```

Comment:
I dont see any need of returning pointer to structure.
```suggestion
func stubVPCMachineTemplate(profile string) infrav1beta2.IBMVPCMachineTemplate {
```

Reply:
Done.